### PR TITLE
🦋 Upgrade all legacy tables with Patternfly

### PR DESCRIPTION
**Pull request by josemigallas**
| page | issue | path | old |
| --- | --- | --- | --- |
| ActiveDocs | https://issues.redhat.com/browse/THREESCALE-9781 | admin/api_docs/services | [Screenshot](https://github.com/3scale/porta/assets/11672286/d57c62a5-d5e1-415c-b1a4-d130a918a3f5) |
| API Alerts | https://issues.redhat.com/browse/THREESCALE-9775 | apiconfig/alerts | [screenshot](https://github.com/3scale/porta/assets/11672286/f1348f74-6587-404b-ba75-14ebdcc794b5) |
| Service contracts | https://issues.redhat.com/browse/THREESCALE-9787 | buyers/service_contracts | [Screenshot](https://github.com/3scale/porta/assets/11672286/aa2e130f-82b9-4f21-b8eb-f6cc6e568ab1) |
| Buyer services subscriptions | https://issues.redhat.com/browse/THREESCALE-9868 | buyers/accounts/:id/service_contracts | [Screenshot](https://github.com/3scale/porta/assets/11672286/e821f079-f2fd-47a4-b96a-b44a82f14672) |
| Audience applications | https://issues.redhat.com/browse/THREESCALE-9967 | p/admin/applications | |
| Product applications | https://issues.redhat.com/browse/THREESCALE-9793 | apiconfig/services/:id/applications | [Screenshot](https://github.com/3scale/porta/assets/11672286/196e7f0c-6a8b-4ee9-ba4c-4d66de375026) |
| Buyer applications | https://issues.redhat.com/browse/THREESCALE-9961 | buyers/accounts/:id/applications | |
| Application overview | https://issues.redhat.com/browse/THREESCALE-9901 | p/admin/applications/:id | [Screenshot](https://github.com/3scale/porta/assets/11672286/ad2dbe45-6e5f-4fe7-b1d1-64041857db00) |
| Application usage rules | https://issues.redhat.com/browse/THREESCALE-9909 | apiconfig/services/:id/usage_rules | [Screenshot](https://github.com/3scale/porta/assets/11672286/98f9e52e-3f9b-4fcd-9312-d74cdb15cf1a) |
| Integration errors | https://issues.redhat.com/browse/THREESCALE-9804 | apiconfig/services/:id/errors | [Table view](https://github.com/3scale/porta/assets/11672286/98c3ef83-b3a4-429b-8eec-83d66fa1bd09), [Empty view](https://github.com/3scale/porta/assets/11672286/7ab3aaeb-274d-4ac8-9155-8f002fae1675) |
| Mapping rules | https://issues.redhat.com/browse/THREESCALE-9810 | apiconfig/services/:id/proxy_rules, p/admin/backend_apis/:id/mapping_rules | [Screenshot](https://github.com/3scale/porta/assets/11672286/ddf5e8ee-c72e-4b46-af60-a61ef06ecdf7) |
| Backend usages | https://issues.redhat.com/browse/THREESCALE-9815 | apiconfig/services/:id/backend_usages | [screenshot](https://github.com/3scale/porta/assets/11672286/a1b4e553-e795-4fa7-8fea-6ef8bac88ba3) |
| Finance dashboard | https://issues.redhat.com/browse/THREESCALE-9819 | /finance | [screenshot](https://github.com/3scale/porta/assets/11672286/a88687f9-bba0-4505-8c9e-e9387cd1490b) |
| Finance logs | https://issues.redhat.com/browse/THREESCALE-9911 | finance/log_entries | [Screenshot](https://github.com/3scale/porta/assets/11672286/a337b5c6-9494-46d1-a8c0-30601b6130f2) |
| Invoices | https://issues.redhat.com/browse/THREESCALE-9822 | finance/invoices | [Screenshot](https://github.com/3scale/porta/assets/11672286/068f2307-dcc2-4cf1-bba2-6e69376210c1) |
| Buyer Invoices | https://issues.redhat.com/browse/THREESCALE-9886 | buyers/accounts/:id/invoices | [Screenshot](https://github.com/3scale/porta/assets/11672286/09f40e0a-2aab-4dd3-a9d8-c88448df664f) |
| Buyer Invoice edit | https://issues.redhat.com/browse/THREESCALE-9907 | buyers/accounts/:id/invoices/:id/edit | [Screenshot](https://github.com/3scale/porta/assets/11672286/54684c95-2609-4c1a-94de-a134bc32db55) |
| Provider Invoices | https://issues.redhat.com/browse/THREESCALE-9895 | p/admin/account/invoices | [Screenshot](https://github.com/3scale/porta/assets/11672286/6a47946e-bd72-49b6-b37c-b2e0c5e03857) |
| Invoice detail | https://issues.redhat.com/browse/THREESCALE-9889 | buyers/accounts/:account_id/invoices/:id, finance/invoices/:id, p/admin/account/invoices | [Screenshot](https://github.com/3scale/porta/assets/11672286/574c04a7-e03b-4844-b1c3-4de48134f4f7) |
| Drafts | https://issues.redhat.com/browse/THREESCALE-9826 | p/admin/cms/changes | [Screenshot](https://github.com/3scale/porta/assets/11672286/7b10cfd0-579d-4b04-9597-17bde84aadff) |
| Redirects | https://issues.redhat.com/browse/THREESCALE-9828 | p/admin/cms/redirects | [Screenshot](https://github.com/3scale/porta/assets/11672286/ceb1cf1b-da46-432e-87f0-92397b9980b1) |
| Groups | https://issues.redhat.com/browse/THREESCALE-9839 | p/admin/cms/groups | [Screenshot](https://github.com/3scale/porta/assets/11672286/e25564ac-00f3-4d22-99b8-55883cb09db1) |
| Feature visibility | https://issues.redhat.com/browse/THREESCALE-9842 | p/admin/cms/switches | [Screenshot](https://github.com/3scale/porta/assets/11672286/e2a3a6f0-6a09-4f62-9461-d18a0329b7ee) |
| Versions | https://issues.redhat.com/browse/THREESCALE-9927 | p/admin/cms/templates/:id/versions | [Screenshot](https://github.com/3scale/porta/assets/11672286/f2125dc8-3b1a-4e2e-9729-f3c94240809b) |
| Account SSO Integrations | https://issues.redhat.com/browse/THREESCALE-9848 | p/admin/account/authentication_providers | [Screenshot](https://github.com/3scale/porta/assets/11672286/e6dc3050-3dcc-40c0-bc2c-8cfa10c19103) |
| Dev Portal SSO Integrations | https://issues.redhat.com/browse/THREESCALE-9850 | p/admin/authentication_providers | [Screenshot](https://github.com/3scale/porta/assets/11672286/29011b25-b088-4164-abc2-204c6f439a04) |
| Messages inbox | https://issues.redhat.com/browse/THREESCALE-9854 | p/admin/messages | [Screenshot](https://github.com/3scale/porta/assets/11672286/9e9700a5-0dee-4f87-85c0-224ff9fbc48c) |
| Messages outbox | https://issues.redhat.com/browse/THREESCALE-9858 | p/admin/messages/outbox | [Screenshot](https://github.com/3scale/porta/assets/11672286/5877c85c-fe95-45cc-9772-0216ee9e09f4) |
| Messages trash | https://issues.redhat.com/browse/THREESCALE-9862 | p/admin/messages/trash | [Screenshot](https://github.com/3scale/porta/assets/11672286/38c87e19-2069-40ce-a765-e0e435cad522) |
| Email templates | https://issues.redhat.com/browse/THREESCALE-9865 | p/admin/cms/email_templates | [Screenshot](https://github.com/3scale/porta/assets/11672286/6a033c98-7429-4874-b24c-e26ddc311446) |
| Personal tokens | https://issues.redhat.com/browse/THREESCALE-9870 | p/admin/user/access_tokens | [Screenshot](https://github.com/3scale/porta/assets/11672286/d986f946-ac8f-460b-b2eb-d1c5e7bf547a) |
| Users listing | https://issues.redhat.com/browse/THREESCALE-9874 | p/admin/account/users | [Screenshot](https://github.com/3scale/porta/assets/11672286/93a02425-801d-4a96-acd7-f21bfd233139) |
| Users Invitations | https://issues.redhat.com/browse/THREESCALE-9877 | p/admin/account/invitations | [Screenshot](https://github.com/3scale/porta/assets/11672286/4f2411db-ba69-49c9-a10a-356d20003b48), [Screenshot](https://github.com/3scale/porta/assets/11672286/21cee24f-1286-477e-912e-8d29c3567d80) |
| Account users invitations | https://issues.redhat.com/browse/THREESCALE-9882 | buyers/accounts/:id/invitations | [Screenshot](https://github.com/3scale/porta/assets/11672286/bcf66494-5c5a-4e77-a95a-c3f69b80ba3b) |
| Account settings overview | https://issues.redhat.com/browse/THREESCALE-9899 | p/admin/account | [Screenshot](https://github.com/3scale/porta/assets/11672286/50360e30-16de-4ff8-a5ef-03442ccd52e1) |
| Change provider plan (modal) | https://issues.redhat.com/browse/THREESCALE-9914 | /buyers/accounts/:id/ | [Screenshot](https://github.com/3scale/porta/assets/11672286/894cfd34-5906-467a-90b0-03dd5cd24b0f) |

### CODE REVIEW

File: app/views/provider/admin/messages/trash/index.html.slim [hunk 1]
```diff
@@ -8,20 +8,24 @@
   // the whole explanation: https://github.com/3scale/system/issues/6275
   //.operations
   //  = fancy_button_to 'Empty the trash', empty_provider_admin_messages_trash_index_path, method: :delete, class: 'empty_trash'
-  table#messages.data
+  table class="data pf-c-table pf-m-grid-lg" role="grid" id="messages"
```
Comment (lvillen) [inline app/views/provider/admin/messages/trash/index.html.slim]: What do you think about adding an `aria-label` here as in `inbox` and `outbox`?

```suggestion
  table class="data pf-c-table pf-m-grid-lg" role="grid" aria-label="Messages trash" id="messages"
```

Comment (josemigallas) [reply app/views/provider/admin/messages/trash/index.html.slim]: I didn't like the label much.. so I skipped it. I am not sure "Messages trash" is meaningful enough.

**Review by jlledom — APPROVED** [review]

File: app/views/admin/api_docs/base/_active_docs_table.slim [hunk 2]
```diff
@@ -1,22 +1,26 @@
...
-    th Swagger Version
-    th class='operations' = link_to('Create a new spec', new_api_docs_service_path(service), class: 'action add')
-  tbody
-    - for api_docs in api_docs_services
-      tr
-        td = link_to api_docs.name, preview_admin_api_docs_service_path(api_docs), title: 'Preview service spec'
-        td = api_docs.system_name
-        td = api_docs.published? ? 'visible' : 'hidden'
+div class="pf-c-card"
+  table class="data pf-c-table pf-m-grid-lg" role="grid" aria-label="Active docs table"
```
Comment (mayorova) [inline app/views/admin/api_docs/base/_active_docs_table.slim:2]: It seems that you've removed `.data` everywhere, but not here.
1. is this intended?
2. is `.data` really useless? (we don't have any case of having multiple tables on the same page?)

**Comment by mayorova** [comment]
I don't like much that the width of the drop-downs is minimally acceptable to hold the content.
E.g.
Screenshot from 2023-07-12 12-52-39

compared to:
Screenshot from 2023-07-12 12-53-05

It looks kind of tight in that box.
Screenshot from 2023-07-12 12-54-02

**Comment by mayorova** [comment]
Applications > Listing pagination doesn't seem to be in the right place:
Screenshot from 2023-07-12 12-55-38

**Comment by mayorova** [comment]
Alerts page:
Screenshot from 2023-07-12 13-09-54



The color coding for the level column has disappeared, is that intentional?

Also, the search input doesn't fit the placeholder text, should it be spanned for another column? I guess it's OK if you think it's not good.

**Comment by mayorova** [comment]
The invoice view is very different:

Screenshot from 2023-07-12 13-16-49

Not sure if that's the intention (the screenshot attached to the PR description shows a different view - I think it's the "as is" version)

**Comment by mayorova** [comment]
When there are no integration error, the message doesn't look cool.

Screenshot from 2023-07-12 14-09-11

compare with "as is":
Screenshot from 2023-07-12 14-09-21

**Comment by mayorova** [comment]
Screenshot from 2023-07-12 14-13-25
I am wondering if this **Purge** button would look better in the last (new?) column, as some other tables have for bulk options (like "Delete All" on Alerts table)

**Comment by mayorova** [comment]
In the account applications listing I think the column distribution is also not optimal...
e.g. here "service" is too wide, and "plan" is too narrow.

Screenshot from 2023-07-12 14-20-31

**Review by lvillen — DISMISSED** [review]

File: app/views/admin/api_docs/base/_active_docs_table.slim (hunk 2 above)
Comment (josemigallas) [reply app/views/admin/api_docs/base/_active_docs_table.slim:2]: I must've have missed this. Besides `.data` being a really bad name for a selector, I replaced it in every JS script with an ID or pf-c-form. I think all `table.data`s were alone. In the few pages with multiple tables other classes are used.

**Comment by mayorova** [comment]
Screenshot from 2023-07-18 12-25-20
"Edit Details" header (is it a header?) and cancel buttons are weirdly placed.

current master (just for reference, as it's also a bit weird design)
Screenshot from 2023-07-18 12-27-07

**Comment by mayorova** [comment]
Screenshot from 2023-07-18 12-52-31
This screen with service-based ActiveDocs does not look too nice, but well, it's as we had before (just less margins), so that's OK I guess.

**Review by lvillen — APPROVED** [review]

**Pull request merged**